x-pack/plugins/stack_connectors/server/connector_types/crowdstrike/crowdstrike.ts

```diff
@@ -0,0 +1,181 @@
+    this.registerSubAction({
+      name: SUB_ACTION.HOST_ACTIONS,
+      method: 'executeHostActions',
+      schema: CrowdstrikeHostActionsParamsSchema,
+    });
+  }
+
+  public async executeHostActions({ alertIds, ...payload }: CrowdstrikeHostActionsParams) {
+    return this.crowdstrikeApiRequest({
+      url: this.urls.hostAction,
+      method: 'post',
+      params: {
+        action_name: payload.command,
+      },
+      data: {
+        ids: payload.ids,
+      },
+      paramsSerializer,
+      responseSchema: CrowdstrikeHostActionsResponseSchema,
+    });
+  }
+
+  public async getAgentDetails(
+    payload: CrowdstrikeGetAgentsParams
+  ): Promise<CrowdstrikeGetAgentsResponse> {
+    return this.crowdstrikeApiRequest({
+      url: this.urls.agents,
+      method: 'GET',
+      params: {
+        ids: payload.ids,
+      },
+      paramsSerializer,
+      responseSchema: CrowdstrikeGetAgentsResponseSchema,
+    });
+  }
+
+  private async getTokenRequest() {
+    const base64encodedData = Buffer.from(
+      this.secrets.clientId + ':' + this.secrets.clientSecret
+    ).toString('base64');
+
+    const response = await this.request<CrowdstrikeGetTokenResponse>({
+      url: this.urls.getToken,
+      method: 'post',
+      headers: {
+        accept: 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded',
+        authorization: 'Basic ' + base64encodedData,
+      },
+      responseSchema: CrowdstrikeGetTokenResponseSchema,
+    });
+    const token = response.data.access_token;
+    if (token) {
+      // Clear any existing timeout
+      clearTimeout(CrowdstrikeConnector.tokenExpiryTimeout);
+
+      // Set a timeout to reset the token after 29 minutes (it expires after 30 minutes)
+      CrowdstrikeConnector.tokenExpiryTimeout = setTimeout(() => {
+        CrowdstrikeConnector.token = null;
+      }, 29 * 60 * 1000);
+    }
+    return token;
+  }
+
+  private async crowdstrikeApiRequest<R extends CrowdstrikeBaseApiResponse>(
+    req: SubActionRequestParams<R>,
+    retried?: boolean
+  ): Promise<R> {
+    try {
+      if (!CrowdstrikeConnector.token) {
+        CrowdstrikeConnector.token = (await this.getTokenRequest()) as string;
+      }
+
+      const response = await this.request<R>({
+        ...req,
+        headers: {
+          ...req.headers,
+          Authorization: `Bearer ${CrowdstrikeConnector.token}`,
+        },
+      });
+
+      // TODO TC: When we have access to API - verify if we catch an error or response.errors
+      // if (response.errors.length) {
+      //   if (response.errors[0].code === 401 && response.errors[0].message === 'access denied, invalid bearer token' && !retried) {
+      //     CrowdstrikeConnector.token = null;
+      //     return this.crowdstrikeApiRequest(req, true);
+      //   }
+      // }
+
+      return response.data;
+    } catch (error) {
+      if (
+        error.code === 401 &&
+        error.message === 'access denied, invalid bearer token' &&
```

Comment:
feels dangerous to be checking the `message` of the error here. are we sure that value will not change?  would checking the `.code` be sufficient?